### ISSUE TITLE
Bearer

### ISSUE THREAD

**Issue by weskhaled** [fluent-ffmpeg/node-fluent-ffmpeg#882]
### Version information

* fluent-ffmpeg version:
* ffmpeg version:
* OS:

I try to add Authorization Bearer in headers
.addOptions('-headers Authorization: Bearer ...)
but always get error 
"Invalid data found when processing input\n"

### Code to reproduce

```js
      ffmpeg(file)
        // .inputOptions('-headers Authorization:' + token)
        // .addOptions('-headers', 'Authorization:Bearer ' + token)
        .addOptions(
          '-headers Authorization: Bearer ...'
        )

```

(note: if the problem only happens with some inputs, include a link to such an input file)

### Checklist

* [ ] I have read the [FAQ](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/wiki/FAQ)
* [ ] I tried the same with command line ffmpeg and it works correctly (hint: if the problem also happens this way, this is an ffmpeg problem and you're not reporting it to the right place)
* [ ] I have included full stderr/stdout output from ffmpeg

**Comment by FullstackJack** [comment]
I still have to try this myself, but I think you forgot to wrap the header in double quotes.
```
.addOptions(
  '-headers "Authorization: Bearer ..."'
)
```

That won't change that fact that it might fail to send those headers if they aren't added before the input. I wonder if fluent will allow that...

**Comment by FullstackJack** [comment]
Yep, just what I thought. Private bucket on Google Cloud Storage fails even though I have provided a Bearer access token as an Authorization header.

`https://www.googleapis.com/storage/v1/b/my-bucket/o/test_file.mp4: Server returned 401 Unauthorized (authorization failed)`

Looks like I'll have to fork the repo and see what can be done. Previously I was building and spawning my own ffmpeg commands, but I'd like to use a community library.

`Server returned 401 Unauthorized (authorization failed)`

**Comment by FullstackJack** [comment]
@weskhaled Fear not, I have a fix on the way. 🏇

**Comment by FullstackJack** [comment]
How does `headers()` sound?

**Comment by FullstackJack** [comment]
I'm beginning to wonder if I should write a new library to specifically tailor to the problem area of HTTP live streaming, thumbnail and GIF generation. This API is convenient for adding individual settings, but you really don't get much in terms of cognitive simplification. You still just set values one-by-one for things you would normally by issuing your own `spawn("ffmpeg", args)`. I would much rather like to see a library derive values from an abstracted higher level configuration based on a known common set of live streaming configurations. Things like choosing a GOP, segment length, bitrates, and scaling should be based on common values with optional overrides. There is too much reinvention of the wheel everytime you start a new live streaming project.

**Comment by FullstackJack** [comment]
@weskhaled I have a PR up that makes it easy to add headers. =)

**Comment by weskhaled** [comment]
thanks ;)

**Comment by njoyard** [comment]
`-headers` is an input option as others have stated. You must use `inputOption` to add it.

```
command.inputOption('-headers', 'Header1: value\r\nHeader2: value...')
```

**Comment by FullstackJack** [comment]
@njoyard I believe inputOptions is missing from ffprobe. Regardless, I think it's a nicer API to have a headers method.

**Comment by njoyard** [comment]
Then it would be nice to add a new channel of options, which would contain options necessary to actually open the input, that could be used in ffprobe as well. Most likely `-headers` is not the only option in this case.

**Comment by snobu** [comment]
Piggybacking on this thread with my own problem -
```
        ffmpeg(`"${video.playbackUrl}"`).inputOption([
            `-headers "Authorization:Bearer ${accessToken}"`
          ])
          ...
```

results in 

```
Stderr output: Unrecognized option 'headers "Authorization:Bearer eyJ[.....]OkPdNOCt5mg"'.
Stderr output: Error splitting the argument list: Option not found
```

Not sure what i'm doing wrong. If i copy paste the command from the `.on('start'..` event, it runs just fine in the console, the video is downloaded without any drama.

Here's what _start_ event gives me -
```
Spawned Ffmpeg with command: ffmpeg -headers "Authorization:Bearer eyJ[...]OCt5mg" -i "https://some.crazy.HLS.url/manifest(format=m3u8-aapl)" -y -f mp4 "videos/output.mp4"
```

**Comment by snobu** [comment]
Nevermind, i just needed to split the array into header and value:

```
 '-headers',  `"Authorization:Bearer ${accessToken}"`
```
